examples/Planning/ForecastReach.cs

```diff
@@ -0,0 +1,313 @@
+        {
+            ListPlannableLocationsRequest request = new ListPlannableLocationsRequest();
+            ListPlannableLocationsResponse response = reachPlanService.ListPlannableLocations(request);
+
+            Console.WriteLine("Plannable Locations:");
+            Console.WriteLine("Name,\tId,\t,ParentCountryId");
+            foreach (PlannableLocation location in response.PlannableLocations)
+            {
+                Console.WriteLine($"\"{location.Name}\",\t{location.Id},{location.ParentCountryId}");
+            }
+        }
+
+
+        /// <summary>
+        /// Lists plannable products for a given location.
+        /// </summary>
+        /// <param name="reachPlanService">Instance of Reach Plan Service client.</param>
+        /// <param name="locationId">Location ID from somewhere returned by ListPlannableLocations.</param>
+        public void ShowPlannableProducts(ReachPlanServiceClient reachPlanService, string locationId)
+        {
+            ListPlannableProductsRequest request = new ListPlannableProductsRequest()
+            {
+                PlannableLocationId = locationId
+            };
+            ListPlannableProductsResponse response = reachPlanService.ListPlannableProducts(request);
+
+            Console.WriteLine($"Plannable Products for location {locationId}:");
+            foreach (ProductMetadata product in response.ProductMetadata)
+            {
+                Console.WriteLine($"{product.PlannableProductCode}:");
+                Console.WriteLine("Age Ranges:");
+                foreach (var ageRange in product.PlannableTargeting.AgeRanges)
+                {
+                    Console.WriteLine($"\t- {ageRange}");
+                }
+                Console.WriteLine("Genders:");
+                foreach (var gender in product.PlannableTargeting.Genders)
+                {
+                    Console.WriteLine($"\t- {gender.Type}");
+                }
+                Console.WriteLine("Devices:");
+                foreach (var device in product.PlannableTargeting.Devices)
+                {
+                    Console.WriteLine($"\t- {device.Type}");
+                }
+            }
+        }
+
+        /// <summary>
+        /// Create a base request to generate a reach forecast.
+        /// </summary>
+        /// <param name="customerId">The customer ID for the reach forecast.</param>
+        /// <param name="productMix">The product mix for the reach forecast.</param>
+        /// <param name="locationId">Location ID from somewhere returned by ListPlannableLocations.</param>
+        /// <param name="currencyCode">Three-character ISO 4217 currency code.</param>
+        public GenerateReachForecastRequest BuildReachRequest(
+                   string customerId,
+                   List<PlannedProduct> productMix,
+                   string locationId,
+                   string currencyCode)
+        {
+            CampaignDuration duration = new CampaignDuration()
```

Comment:
Please document if these values have any special relevance (e.g. 30 days). Point to relevant Ads HC article when applicable.

Reply:
Added a comment -- it doesn't look like we put valid ranges in the reference docs so I'll make sure we push a corresponding change to the proto comments.